Feature/ab#33 participant list export

wim07101993 opened the pull request:
Added possibility to download a csv file containing the data of the registrations on an event.

wim07101993 committed:
feature/particicipant-export: created csv methods and attributes

AB#33

wim07101993 committed:
feature/participant-export: created unit tests for csv serialization

wim07101993 committed:
feature/participant-export: implemented CsvHelper nuget

AB#33

wim07101993 committed:
feature/participant-export: created and implemented button to download csv file

AB#33

wim07101993 committed:
feature/participant-export: prettier file-name

AB#33

wim07101993 committed:
Merge branch 'develop' into feature/AB#33-participant-list-export

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json
#	Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singer-event-registrations/singer-event-registrations.component.ts
#	Singer.API/ClientApp/src/app/modules/core/services/singerevents-api/singerevents.proxy.ts
#	Singer.API/ClientApp/src/app/modules/core/services/singerevents-api/singerevents.service.ts

wim07101993 committed:
feature/participant-export: changed csv properties to match request from StG

AB#33

vyruz1986 commented on Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singer-event-registrations/singer-event-registrations.component.ts:
```diff
@@ -168,4 +169,36 @@ export class SingerEventRegistrationsComponent implements OnInit {
+               );
+            },
+            error => {
+               console.log('Error downloading the file');
```
Please remove console logs and replace with SnackBar notifications

vyruz1986 commented on Singer.API/DTOs/Csv/CsvRegistrationDTO.cs:
```diff
@@ -0,0 +1,36 @@
+      {
+         public Mapper()
+         {
+            Map(x => x.CaseNumber).Index(0).Name(DisplayNames.CaseNumber);
```
Why don't we use automapper here?

wim07101993 replied:
Because this mapper is not for db, but for csv. It is a way to ensure the properties are in a certain order and a certain the name, format,.. is used during the serialization

vyruz1986 commented on Singer.API/Helpers/Attributes/CsvPropertyAttribute.cs:
```diff
@@ -0,0 +1,19 @@
+         _propertyName = propertyName;
+      }
+
+      public string PropertyName => _propertyName ?? PropertyNameResourceType?.GetProperty(PropertyNmeResourceName)?.GetValue(null, null) as string;
```
😎

wim07101993 replied:
whoops, those files can be deleted...

vyruz1986 commented on Singer.API/Services/EventRegistrationService.cs:
```diff
@@ -197,6 +199,28 @@ public async Task<EventRegistrationDTO> GetOneAsync(Guid eventId, Guid registrat
+               AgeGroup = x.CareUser.AgeGroup,
+               DayCareAfterEndDateTime = x.EventSlot.Event.DayCareAfterEndDateTime,
+               DayCareBeforeStartDateTime = x.EventSlot.Event.DayCareBeforeStartDateTime,
+               LegalGuardianName = "NOT IMPLEMENTED YET",
```
Add TODO please so we don't forget

vyruz1986 commented on Singer.API/appsettings.Development.json:
```diff
@@ -7,7 +7,7 @@
    },
    "ConnectionStrings": {
-      "Application": "Server=localhost,1433;Database=Singer;User=sa;Password=singer-2019-password;"
+      "Application": "Server=192.168.56.101,1433;Database=Singer;User=sa;Password=singer-2019-password;"
```
whut? change it back!

wim07101993 replied:
...

wim07101993 committed:
feature/participant-export: fixed remarks pull request:
- added TODO to csv DTO mapping
- changed console log to snackbar
- removed useless attributes

AB#33

wim07101993 committed:
Merge branch 'develop' into feature/AB#33-participant-list-export

wim07101993 committed:
feature/participant-export: fixed build

vyruz1986 commented on Singer.API/Services/EventRegistrationService.cs:
```diff
@@ -197,6 +199,28 @@ public async Task<EventRegistrationDTO> GetOneAsync(Guid eventId, Guid registrat
+               AgeGroup = x.CareUser.AgeGroup,
+               DayCareAfterEndDateTime = x.EventSlot.Event.DayCareAfterEndDateTime,
+               DayCareBeforeStartDateTime = x.EventSlot.Event.DayCareBeforeStartDateTime,
+               LegalGuardianName = "TODO add legal guardian",
```
I mean `//TODO Remember to do this`

wim07101993 replied:
Oooh...

wim07101993 committed:
feature/participant-export: Updated TODO legal guardian

AB#33

wim07101993 committed:
feature/participant-export: modified and added TODO to CsvRegistrationDTO

AB#33

wim07101993 committed:
Merge branch 'develop' into feature/AB#33-participant-list-export

wim07101993 committed:
Merge branch 'develop' into feature/AB#33-participant-list-export

vyruz1986 committed:
Merge branch 'develop' into feature/AB#33-participant-list-export
